docs/details/develop-checks.md

```diff
@@ -1,266 +1,313 @@
+All checks are self-contained in the first phase, as well as in most of the planned phases.
+ 
+This means that extracted data is processed on the PMM side and not sent back to the SaaS.
 
-- SQL query or MongoDB query document for extracting data from the database.
-- Python script for converting extracted data into check results. This is actually a [Starlark](https://github.com/google/starlark-go) script – a Python dialect that adds more imperative features from the Python. Script's execution environment is sandboxed; no I/O can be done from it.
-
-All checks in the first phase (and most of the planned ones) are self-contained. This means that extracted data is processed on the PMM side and not sent back to the SaaS.
-
-On the other hand, checks results and other metadata can be sent to SaaS to implement a history feature if the user opted-in and is entitled.
+## Format
 
-For example, below is a single check that returns the static result:
-
-```yaml
+``` yaml
 ---
 checks:
-  - version: 1
-    name: example
-    summary: Example check
-    description: This check is just an example.
-    tiers: [anonymous, registered]
-    type: MONGODB_BUILDINFO
-    script: |
-      def check(docs):
-          # for compatibility with PMM Server < 2.12
-          context = {
-              "format_version_num": format_version_num,
-              "parse_version": parse_version,
-          }
-          return check_context(docs, context)
-
-
-      def check_context(docs, context):
-          # `docs` is a frozen (deeply immutable) list of dicts where each dict represents a single document in result set.
-          # `context` is a dict with additional functions.
-          #
-          # Global `print` and `fail` functions are available.
-          #
-          # `check_context` function is expected to return a list of dicts that are then converted to alerts;
-          # in particular, that list can be empty.
-          # Any other value (for example, string) is treated as script execution failure
-          # (Starlark does not support Python exceptions);
-          # it is recommended to use global function `fail` for that instead.
-
-          format_version_num = context.get("format_version_num", fail)
-          parse_version = context.get("parse_version", fail)
-
-          print("first doc =", repr(docs[0]))
-
-          return [{
-              "summary": "Example summary",
-              "description": "Example description",
-              "severity": "warning",
-              "labels": {
-                  "version": format_version_num(10203),
-              }
-          }]
-```
+  - version: 2             <------ version increment
+    name: exampleV2
+    summary: Check format V2
+    description: Checks something important
+    interval: standard
+    family: MYSQL          <------- family
+    queries:               <-------- queries
+      - type: MYSQL_SELECT
+        query: some query
+
+      - type: MYSQL_SHOW
+        query: some query 
```

Comment:
```suggestion
  - version: 1             <------ version increment
    name: exampleV1
    summary: Check format V1
    description: Checks something important
    interval: standard
    type: MYSQL_SELECT
    query:                   <-------- query
```

Reply:
V2 is not available yet.

Reply:
I think we can take some simple check from `anonymous` tier and use it as an example with comments to show the format.